mkt/extensions/models.py

```diff
@@ -33,96 +34,62 @@
 
 class ExtensionManager(ManagerBase):
     def pending(self):
-        return self.filter(status=STATUS_PENDING).order_by('id')
+        return self.filter(versions__status=STATUS_PENDING).order_by('id')
 
     def public(self):
         return self.filter(status=STATUS_PUBLIC).order_by('id')
 
 
 class Extension(ModelBase):
+    # Automatically handled fields.
     uuid = UUIDField(auto=True)
+    status = models.PositiveSmallIntegerField(
+        choices=STATUS_CHOICES.items(), db_index=True, default=STATUS_NULL)
 
     # Fields for which the manifest is the source of truth - can't be
     # overridden by the API.
     default_language = models.CharField(default=settings.LANGUAGE_CODE,
                                         max_length=10)
-    manifest = JSONField()
-    version = models.CharField(max_length=255, default='')
+    description = TranslatedField(default=None)
+    name = TranslatedField(default=None)
 
     # Fields that can be modified using the API.
     authors = models.ManyToManyField('users.UserProfile')
-    name = TranslatedField(default=None)
     slug = models.CharField(max_length=35, unique=True)
-    status = models.PositiveSmallIntegerField(
-        choices=STATUS_CHOICES.items(), db_index=True, default=STATUS_NULL)
 
     objects = ExtensionManager()
 
-    def clean_slug(self):
-        return clean_slug(self, slug_field='slug')
+    @cached_property(writable=True)
+    def latest_public_version(self):
+        return self.versions.filter(status=STATUS_PUBLIC).latest('pk')
```

Comment:
either way is good for me. pk may be a bit faster (almost negligibly faster), although created semantically makes more sense